src/clients/HubPoolClient.ts

```diff
@@ -254,10 +258,39 @@ export class HubPoolClient extends BaseAbstractClient {
     l1Token: string,
     blockNumber: number,
     amount: BigNumber,
-    // eslint-disable-next-line @typescript-eslint/no-unused-vars
-    _timestamp: number
+    timestamp: number
   ): Promise<{ current: BigNumber; post: BigNumber }> {
-    return this.getPostRelayPoolUtilization(l1Token, blockNumber, amount);
+    // Resolve this function call as an async anonymous function
+    // This way, since we have to use this call several times, we
+    // only need to invoke the shorter function name.
+    const resolver = async () => this.getPostRelayPoolUtilization(l1Token, blockNumber, amount);
+    // Resolve the cache locally so that we can appease typescript
+    const cache = this.cachingMechanism;
+    // If there is no cache, just resolve the function
+    if (!cache) {
+      return resolver();
+    }
+    // Otherwise, let's resolve the key
+    const key = `utilization_${l1Token}_${blockNumber}_${amount.toString()}`;
+    // Resolve the key from the cache
+    const result = await cache.get<string>(key);
+    // We were able to find a valid result, so let's return it
+    if (isDefined(result)) {
+      const [current, post] = result.split(",").map(BigNumber.from);
+      return { current, post };
+    }
+    // We were not able to find a valid result, so let's resolve the function
+    // and store the result in the cache
+    else {
+      const { current, post } = await resolver();
+      // First determine if we should cache the result
+      if (shouldCache(getCurrentTime(), timestamp, 60 * 60 * 24 * 90)) {
```

Comment:
So just to confirm, you are saying we should only cache utilization data that is at least 90 days old?

Reply:
Correct - I've modified it to at least one day old. This should cache more items while reducing reorg risk